### PR TITLE
KIALI-2640 Add support for Service definition updates

### PR DESCRIPTION
**Pull request by lucasponce**
In the context of third-party integration with adapters (3scale on this case), as part of the process of integration, some istio objects needs to be created, but also additional labels need to be added at service definition level.

In order to perform that operation, we need to enable a PATCH operation in Kiali for service defintions.

https://issues.jboss.org/browse/KIALI-2640

How to test:

```
## To add a new label into an existing service

PATCH http://admin:admin@localhost:8000/api/namespaces/bookinfo/services/details
Content-Type: application/json

{
  "metadata": {
    "labels": {
      "3scale": "bookinfo-3scale.3scale.net"
    }
  }
}

## To remove an existing field, this needs to be set as null
## Additional info about json merge patch format used by k8s https://tools.ietf.org/html/rfc7386#page-8

```

### CODE REVIEW

**Comment by rhqci** [comment]
Jenkins CI: kiali-core-pr-e2e-test [#838](https://paste.fedoraproject.org/paste/GJSVn-qtfNsHtoGdYdcofw)
  * :heavy_check_mark: run-kiali-e2e-tests [#[1660]](https://paste.fedoraproject.org/paste/hQvp4Sq~G44KGJe1xMaDIw)

**Comment by rhqci** [comment]
Jenkins CI: kiali-core-pr-e2e-test [#839](https://paste.fedoraproject.org/paste/aKoTaJKaoHW1DPgKxw-WKg)
  * :heavy_check_mark: run-kiali-e2e-tests [#[1661]](https://paste.fedoraproject.org/paste/22XNjmB~fORf~JAAmGw3Vw)

**Comment by lucasponce** [comment]
cc @jmazzitelli @gbaufake 
This requires changes on clusterrole.xml as well

**Comment by lucasponce** [comment]
https://issues.jboss.org/browse/MAISTRA-358

File: handlers/services.go [hunk 1]
```diff
@@ -130,6 +131,40 @@ func ServiceDetails(w http.ResponseWriter, r *http.Request) {
...
+		return
+	}
+
+	params := mux.Vars(r)
+	namespace := params["namespace"]
+	service := params["service"]
+
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		RespondWithError(w, http.StatusBadRequest, "Update request with bad update patch: "+err.Error())
```
Comment (jotak) [inline handlers/services.go]: I guess a `return` is missing here?

Comment (lucasponce) [reply handlers/services.go]: right, and probably from the place I copied :-) too.

**Comment by rhqci** [comment]
Jenkins CI: kiali-core-pr-e2e-test [#845](https://paste.fedoraproject.org/paste/GXaahgDMbVHjdKZWLwoDpA)
  * :heavy_check_mark: run-kiali-e2e-tests [#[1665]](https://paste.fedoraproject.org/paste/Gl5mLSWeMyJwmAJ2hxd2fA)

**Comment by lucasponce** [comment]
I temporarily close this. Revisiting some doc, I guess I missed to enable the patch verb at workload level.
So, won't merge this yet until I can confirm it.

**Pull request closed without merging**